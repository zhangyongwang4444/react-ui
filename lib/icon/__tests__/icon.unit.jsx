import renderer from 'react-test-renderer'
import React from 'react'
import {mount} from 'enzyme'
import Icon from '../icon'

describe('icon', () => {
    it('render success', () => {
        const json = renderer.create(<Icon name="wechat"/>).toJSON();
        expect(json).toMatchSnapshot()
    });

    it('onClick', () => {
        let n = 1;
        const fn = () => {
            n = 2;
        };
        const component = mount(<Icon name="wechat" onClick={fn}/>);
        component.find('svg').simulate('click');
        expect(n).toEqual(2);
    })
});



