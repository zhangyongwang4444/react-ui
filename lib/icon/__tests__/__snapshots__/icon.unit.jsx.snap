// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render success 1`] = `
<span>
  <svg
    className="react-ui-icon"
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;

exports[`icon 是个 svg 1`] = `
<span>
  <svg
    className="react-ui-icon"
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
